presto-orc/src/main/java/io/prestosql/orc/metadata/statistics/TimestampStatisticsBuilder.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.orc.metadata.statistics;
+
+import java.util.List;
+import java.util.Optional;
+
+import static io.prestosql.orc.metadata.statistics.IntegerStatistics.INTEGER_VALUE_BYTES;
+import static java.util.Objects.requireNonNull;
+
+public class TimestampStatisticsBuilder
+        implements LongValueStatisticsBuilder
+{
+    private long nonNullValueCount;
+    private long minimum = Long.MAX_VALUE;
+    private long maximum = Long.MIN_VALUE;
+
+    @Override
+    public void addValue(long value)
+    {
+        nonNullValueCount++;
+
+        minimum = Math.min(value, minimum);
+        maximum = Math.max(value, maximum);
+    }
+
+    private void addIntegerStatistics(long valueCount, IntegerStatistics value)
```

Comment:
Shouldn't this be `private void addTimestampStatistics(long valueCount, TimestampStatistics value)`?